drivers/SmartThings/zigbee-button/src/zigbee-multi-button/wallhero/init.lua

```diff
@@ -0,0 +1,103 @@
+-- Copyright 2023 SmartThings
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+
+local capabilities = require "st.capabilities"
+local log = require "log"
+local socket = require "cosock.socket"
+local zcl_clusters = require "st.zigbee.zcl.clusters"
+
+local Scenes = zcl_clusters.Scenes
+
+local FINGERPRINTS = {
+  { mfr = "WALL HERO", model = "A4" }
+}
+
+local function can_handle_wallhero_button(opts, driver, device, ...)
+  for _, fingerprint in ipairs(FINGERPRINTS) do
+    if device:get_manufacturer() == fingerprint.mfr and device:get_model() == fingerprint.model then
+      return true
+    end
+  end
+  return false
+end
+
+local function scenes_cluster_handler(driver, device, zb_rx)
+  local additional_fields = {
+    state_change = true
+  }
+
+  local ep = zb_rx.address_header.src_endpoint.value
+  local button_name = "button" .. ep
+  local event = capabilities.button.button.pushed(additional_fields)
+  local comp = device.profile.components[button_name]
+  if comp ~= nil then
+    device:emit_component_event(comp, event)
```

Comment:
this is fine, but in the future you can save yourself a line and just use `emit_event_for_endpoint` which goes and calls `endpoint_to_component`